File: Holiday Assignments/Java/bookSuggestionSystem.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.Scanner;

public class BookSuggestionSystem {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        Random random = new Random();

        

    List<String> bookList = new ArrayList<>();
    bookList.add("Brave King");
    bookList.add("The Hobbit");
    bookList.add("The Mystery");
    bookList.add("Animal Farm");

    boolean continueSuggestion = true;

    mainloop:
    while (true){
                System.out.println(
                        """
                        Welcome to the Book Suggestion System!
                        1. Get Suggestions
                        2. Add Book
                        3. Remove Book
                        4. Update book
                        5. Show all books
                        0. Exit

                        Enter here: """
                        );

                int bookSelectionInput = input.nextInt();
                input.nextLine();
                
                switch (bookSelectionInput){
                    case 1 -> {
                        while (continueSuggestion == true){

                            System.out.println("Get Suggestions selected");
                            System.out.println("Book of the day:");
                            int bookIndex = random.nextInt(bookList.size());
                            int pageNumber = random.nextInt(100) + 1;

                            System.out.println("Book Title: " + bookList.get(bookIndex));
                            System.out.println("Page: " + pageNumber);
                            System.out.print("Would you like to get another suggestion? (yes/no): ");
                            String continueInput = input.nextLine().toLowerCase();
                            
                            if (continueInput.equals("no")){
                                 continueSuggestion = false;
                            }
                            else if (continueInput.equals("yes")){
                                continueSuggestion = true;
                            }
                            else{
                                 System.out.println("Invalid input");
				 break;
                                }
                            }
                            System.out.println();
                        }
                    

                    case 2 -> {
                        System.out.println("\nAdd Book selected.");
                        System.out.print("Book Title: ");
                        String bookAdditionNameInput = input.nextLine();

                        if (bookList.contains(bookAdditionNameInput)) {
                        System.out.println("Book already here");
                    } 
                        else {
                        bookList.add(bookAdditionNameInput);
                        System.out.println("Book added successfully");
                    }
                    System.out.println();
                }
                     
                    case 3 -> {
                        System.out.println("\nRemove Book selected.");
                        System.out.print("Enter the book title to remove: ");
                        String bookRemovalNameInput = input.nextLine();
                    
                        if (bookList.contains(bookRemovalNameInput)) {
                            bookList.remove(bookRemovalNameInput);
                            System.out.println("Book removed successfully!");
                        } else {
                            System.out.println("Book not here");
                        }
                        System.out.println();
                    }

                    case 4 -> {
                        System.out.println("\nUpdate Book selected.");
                        System.out.print("Enter the old title: ");
                        String bookUpdateNameInputOld = input.nextLine();
                        System.out.print("Enter the new title: ");
                        String bookUpdateNameInputNew = input.nextLine();
                        
                        if (bookList.contains(bookUpdateNameInputOld)) {
                            int index = bookList.indexOf(bookUpdateNameInputOld);
                            bookList.set(index, bookUpdateNameInputNew);
                            System.out.println("Book updated successfully!");
                        } else {
                            System.out.println("Book not found");
                        }
                        System.out.println();
                    }
                    
                    case 5 -> {
                        System.out.println("Show all books selected.");
                        for (int index = 0; index < bookList.size(); index++) {
                            System.out.println((index + 1) + ". " + bookList.get(index));
                        }
                        System.out.println();
                    }
                
                    case 0 -> {
                        System.out.println("Ending program");
                        input.close();
                        break mainloop;
                    }
                    default -> {
                        System.out.println("Invalid, Please try again.");
			System.out.pritnln(" ")
                    }
                }
        }
    }
}

File: Holiday Assignments/Python/book_suggestion_system.py
import random

def bookSuggesstionSystem():
    book_list = ["Brave king", "The Hobbit", "The Mystery", "Animal Farm"]
    continue_suggestion = True
    while True:
                print(
"""
Welcome to the Book Suggestion System!
1. Get Suggestions
2. Add Book
3. Remove Book
4. Update book
5. Show all books
0. Exit

Enter here: """
                        )

                book_selection_input = int(input())
                match book_selection_input:
                    case 1 :
                        while continue_suggestion == True:

                            print("Get Suggestions selected")
                            print("Book of the day:")
                            book_index = random.randint(0, len(book_list) - 1)
                            page_number = random.randint(1, 100)
                            print("Book Title: ",book_list[book_index])
                            print("Page: ", page_number)
                            continue_input = input("Would you like to get another suggestion? (yes/no): ").lower()
                            
                            if continue_input == "no":
                                 continue_suggestion = False
                                 break
                            elif continue_input == "yes":
                                continue_suggestion = True
                                continue
                            else:
                                 print("Invalid input")
                                 break

                    case 2 :
                        print("Add Book selected.")
                        book_addition_name_input = input("Book Title: ").title()

                        if book_addition_name_input in book_list:
                             print("Book already here")
                        else:
                            book_list.append(book_addition_name_input)
                            print("Book added successfully")
                     
                    case 3 :
                        print("Remove Book selected.")
                        book_removal_name_input = input("Enter the book title to remove: ")
                        
                        if book_removal_name_input in book_list:
                             book_list.remove(book_removal_name_input)
                             print("Book removed sucessfully!")
                        else:
                             print("Book not here")

                    case 4 :
                        print("Update Book selected.")
                        book_update_name_input_old = input("Enter the old title: ")
                        book_update_name_input_new = input("Enter the new title: ")

                        if book_update_name_input_old in book_list:
                             index = book_list.index(book_update_name_input_old)
                             book_list[index] = book_update_name_input_new
                             print("Book updated successfully!")
                        else:
                             print("Book not found")
                     
                    case 5 :
                        print("Show all books selected.")
                        for index in range (len(book_list)):
                             print (index + 1, ". ", book_list[index], sep = "")

                    case 0 :
                        print("Ending program")
                        return
                    case _ :
                        print("Invalid, Please try again.")

bookSuggesstionSystem()